libcfnet/misc.c

```diff
@@ -103,12 +103,14 @@ int sockaddr_AddrCompare(const void *sa1, const void *sa2)
         struct in_addr *addr1 = & ((struct sockaddr_in *) sa1)->sin_addr;
         struct in_addr *addr2 = & ((struct sockaddr_in *) sa2)->sin_addr;
         result = memcmp(addr1, addr2, sizeof(*addr1));
+        break;
     }
     case AF_INET6:
     {
         struct in6_addr *addr1 = & ((struct sockaddr_in6 *) sa1)->sin6_addr;
         struct in6_addr *addr2 = & ((struct sockaddr_in6 *) sa2)->sin6_addr;
         result = memcmp(addr1, addr2, sizeof(*addr1));
+        break;
```

Comment:
@vpodzime Did you look at this? Fix seems correct, but seems like a change in behavior. Further, the fact that it was falling through to `assert(0)` indicates that it's not covered by tests.

Reply:
It looked to me like a bug which should be fixed by this. But hard to say for sure, actually.

Reply:
@vpodzime @KotzaBoss This function was introduced here:

https://github.com/cfengine/core/commit/8bf5633b868e2e326ae0feb5b15e15ed545bd9f6

And never used.

@KotzaBoss after this PR is merged, please open another one where you remove this function. You can also remove the other function above, `sockaddr_port()` which is also not used.

Reply:
https://github.com/cfengine/core/pull/4607